include/seqan3/argument_parser/argument_parser.hpp

```diff
@@ -629,7 +656,16 @@ class argument_parser
             }
         }
 
-        format = detail::format_parse(argc, std::move(argv_new));
+        if (!special_format_was_set)
+        {
+            if (!subcommands.empty() && sub_parser == nullptr)
```

Comment:
```suggestion
            if (!subcommands.empty())
```
And remove the check above.

Then this implies that you only reach this point if there was no special format and no subcommand was set.